File: C# Net/Project/FieldGroup.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Testing.ValuesGenerator
{
    public class FieldGroup
    {
        public readonly string FieldName;
        public readonly FieldType FieldType;
        public readonly List<Object> Values;

        public FieldGroup(string fieldName, FieldType fieldType, params Object[] values)
        {
            FieldName = fieldName;
            FieldType = fieldType;
            Values = values.Distinct().ToList();
        }

        public List<List<FieldGroup>> SplitValues()
        {
            return Values.Select(value => new List<FieldGroup> { new FieldGroup(FieldName, FieldType, value) }).ToList();
        }

        public List<List<FieldGroup>> AddCombinations(List<List<FieldGroup>> combinations)
        {
            return Values.SelectMany(value =>
                combinations.Select(listValues =>
                {
                    var result = new List<FieldGroup> { new FieldGroup(FieldName, FieldType, value) };
                    result.AddRange(listValues);
                    return result;
                }).ToList()).ToList();
        }
    }
    public enum FieldType { Field, Property }
}

File: Java/src/main/java/ru/viqa/test_data_generator/generator/FieldGroup.java
package ru.viqa.test_data_generator.generator;

import java.util.ArrayList;
import java.util.List;

import static java.util.Arrays.asList;

/**
 * Created by 12345 on 16.12.2014.
 */
public class FieldGroup {
    public String fieldName;
    public List<Object> values;

    public FieldGroup(String fieldName, Object... values)
    {
        this.fieldName = fieldName;
        this.values = asList(values);
    }

    public List<List<FieldGroup>> splitValues()
    {
        List<List<FieldGroup>> result = new ArrayList<>();
        for (Object value : values) {
            List<FieldGroup> el = new ArrayList<>();
            el.add(new FieldGroup(fieldName, value));
            result.add(el);
        }
        return result;
    }

    public List<List<FieldGroup>> addCombinations(List<List<FieldGroup>> combinations)
    {
        List<List<FieldGroup>> result = new ArrayList<>();
        for (Object value : values)
            for (List<FieldGroup> listValues : combinations) {
                List<FieldGroup> list = new ArrayList<>();
                list.add(new FieldGroup(fieldName, value));
                list.addAll(listValues);
                result.add(list);
            }
        return result;
    }
}

